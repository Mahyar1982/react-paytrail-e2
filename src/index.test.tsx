import React from 'react'
import renderer from 'react-test-renderer'

import { Form } from '.'

const getURLs = (baseURI: string): URLSet => ({
  success: new URL(`${baseURI}/success`),
  cancel: new URL(`${baseURI}/cancel`),
  notify: new URL(`${baseURI}/notify`)
})

test('Form renders with minimum fields', async () => {
  const merchant: Merchant = {
    id: '1',
    secret: '123'
  }

  const orderNumber = '1'
  const urls = getURLs('https://example.com')

  const customer: Customer = {
    firstName: 'Alice',
    lastName: 'Angel'
  }

  const product: Product = {
    id: '123',
    title: 'Test Product',
    price: 100
  }

  const form = renderer.create(
    <Form
      merchant={merchant}
      orderNumber={orderNumber}
      customer={customer}
      products={[product]}
      urls={urls}
    />
  )

  const tree = form.toJSON()
  expect(tree).toMatchSnapshot()
})
