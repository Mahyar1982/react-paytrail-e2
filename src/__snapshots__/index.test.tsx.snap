// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders with all the fields 1`] = `
<div>
  <section
    class="e2-form-wrapper"
  >
    <form
      action="https://payment.paytrail.com/e2"
      method="post"
    >
      <input
        name="MERCHANT_ID"
        type="hidden"
        value="1"
      />
      <input
        name="ORDER_NUMBER"
        type="hidden"
        value="1"
      />
      <input
        name="PARAMS_IN"
        type="hidden"
        value="MERCHANT_ID,ORDER_NUMBER,PARAMS_IN,LOCALE,CURRENCY,REFERENCE_NUMBER,EXPIRATION_FOR_PAYMENT_CREATION,PAYMENT_METHODS,VAT_IS_INCLUDED,ALG,URL_SUCCESS,URL_CANCEL,URL_NOTIFY,ITEM_TITLE[0],ITEM_ID[0],ITEM_QUANTITY[0],ITEM_UNIT_PRICE[0],ITEM_VAT_PERCENT[0],ITEM_DISCOUNT_PERCENT[0],ITEM_TYPE[0],PAYER_PERSON_PHONE,PAYER_PERSON_EMAIL,PAYER_PERSON_FIRSTNAME,PAYER_PERSON_LASTNAME,PAYER_COMPANY_NAME,PAYER_PERSON_ADDR_STREET,PAYER_PERSON_ADDR_POSTAL_CODE,PAYER_PERSON_ADDR_TOWN,PAYER_PERSON_ADDR_COUNTRY,MSG_UI_MERCHANT_PANEL,MSG_SETTLEMENT_PAYER,MSG_UI_PAYMENT_METHOD,PARAMS_OUT"
      />
      <input
        name="LOCALE"
        type="hidden"
        value="fi_FI"
      />
      <input
        name="CURRENCY"
        type="hidden"
        value="EUR"
      />
      <input
        name="REFERENCE_NUMBER"
        type="hidden"
        value="RF123456789"
      />
      <input
        name="EXPIRATION_FOR_PAYMENT_CREATION"
        type="hidden"
        value="2021-01-01"
      />
      <input
        name="PAYMENT_METHODS"
        type="hidden"
        value="1,2"
      />
      <input
        name="VAT_IS_INCLUDED"
        type="hidden"
        value="1"
      />
      <input
        name="ALG"
        type="hidden"
        value="1"
      />
      <input
        name="URL_SUCCESS"
        type="hidden"
        value="https://example.com/success"
      />
      <input
        name="URL_CANCEL"
        type="hidden"
        value="https://example.com/cancel"
      />
      <input
        name="URL_NOTIFY"
        type="hidden"
        value="https://example.com/notify"
      />
      <input
        name="ITEM_TITLE[0]"
        type="hidden"
        value="Test Product"
      />
      <input
        name="ITEM_ID[0]"
        type="hidden"
        value="123"
      />
      <input
        name="ITEM_QUANTITY[0]"
        type="hidden"
        value="1"
      />
      <input
        name="ITEM_UNIT_PRICE[0]"
        type="hidden"
        value="100.00"
      />
      <input
        name="ITEM_VAT_PERCENT[0]"
        type="hidden"
        value="24"
      />
      <input
        name="ITEM_DISCOUNT_PERCENT[0]"
        type="hidden"
        value="10"
      />
      <input
        name="ITEM_TYPE[0]"
        type="hidden"
        value="1"
      />
      <input
        name="PAYER_PERSON_PHONE"
        type="hidden"
        value="040123456"
      />
      <input
        name="PAYER_PERSON_EMAIL"
        type="hidden"
        value="alice@example.org"
      />
      <input
        name="PAYER_PERSON_FIRSTNAME"
        type="hidden"
        value="Alice"
      />
      <input
        name="PAYER_PERSON_LASTNAME"
        type="hidden"
        value="Angel"
      />
      <input
        name="PAYER_COMPANY_NAME"
        type="hidden"
        value="Acme Ltd"
      />
      <input
        name="PAYER_PERSON_ADDR_STREET"
        type="hidden"
        value="Test Street 1"
      />
      <input
        name="PAYER_PERSON_ADDR_POSTAL_CODE"
        type="hidden"
        value="00100"
      />
      <input
        name="PAYER_PERSON_ADDR_TOWN"
        type="hidden"
        value="Helsinki"
      />
      <input
        name="PAYER_PERSON_ADDR_COUNTRY"
        type="hidden"
        value="FI"
      />
      <input
        name="MSG_UI_MERCHANT_PANEL"
        type="hidden"
        value="This is a message for the Merchant's Panel"
      />
      <input
        name="MSG_SETTLEMENT_PAYER"
        type="hidden"
        value="This is a message for the customer"
      />
      <input
        name="MSG_UI_PAYMENT_METHOD"
        type="hidden"
        value="This is a message for the payment method"
      />
      <input
        name="PARAMS_OUT"
        type="hidden"
        value="ORDER_NUMBER,PAYMENT_ID,AMOUNT,CURRENCY,PAYMENT_METHOD,TIMESTAMP,STATUS"
      />
      <input
        name="AUTHCODE"
        type="hidden"
        value="24442E884055706A8954DB39EEE246EBD01C92CB4205A1A18CD88A32887389A6"
      />
      <button
        type="submit"
      >
        Pay
      </button>
    </form>
  </section>
</div>
`;

exports[`Form renders with minimum required fields 1`] = `
<div>
  <section
    class="e2-form-wrapper"
  >
    <form
      action="https://payment.paytrail.com/e2"
      method="post"
    >
      <input
        name="MERCHANT_ID"
        type="hidden"
        value="1"
      />
      <input
        name="ORDER_NUMBER"
        type="hidden"
        value="1"
      />
      <input
        name="PARAMS_IN"
        type="hidden"
        value="MERCHANT_ID,ORDER_NUMBER,PARAMS_IN,LOCALE,CURRENCY,VAT_IS_INCLUDED,ALG,URL_SUCCESS,URL_CANCEL,URL_NOTIFY,ITEM_TITLE[0],ITEM_ID[0],ITEM_QUANTITY[0],ITEM_UNIT_PRICE[0],ITEM_VAT_PERCENT[0],ITEM_DISCOUNT_PERCENT[0],ITEM_TYPE[0],PAYER_PERSON_FIRSTNAME,PAYER_PERSON_LASTNAME,PARAMS_OUT"
      />
      <input
        name="LOCALE"
        type="hidden"
        value="fi_FI"
      />
      <input
        name="CURRENCY"
        type="hidden"
        value="EUR"
      />
      <input
        name="VAT_IS_INCLUDED"
        type="hidden"
        value="1"
      />
      <input
        name="ALG"
        type="hidden"
        value="1"
      />
      <input
        name="URL_SUCCESS"
        type="hidden"
        value="https://example.com/success"
      />
      <input
        name="URL_CANCEL"
        type="hidden"
        value="https://example.com/cancel"
      />
      <input
        name="URL_NOTIFY"
        type="hidden"
        value="https://example.com/notify"
      />
      <input
        name="ITEM_TITLE[0]"
        type="hidden"
        value="Test Product"
      />
      <input
        name="ITEM_ID[0]"
        type="hidden"
        value="123"
      />
      <input
        name="ITEM_QUANTITY[0]"
        type="hidden"
        value="1"
      />
      <input
        name="ITEM_UNIT_PRICE[0]"
        type="hidden"
        value="100.00"
      />
      <input
        name="ITEM_VAT_PERCENT[0]"
        type="hidden"
        value="24"
      />
      <input
        name="ITEM_DISCOUNT_PERCENT[0]"
        type="hidden"
        value="0"
      />
      <input
        name="ITEM_TYPE[0]"
        type="hidden"
        value="1"
      />
      <input
        name="PAYER_PERSON_FIRSTNAME"
        type="hidden"
        value="Alice"
      />
      <input
        name="PAYER_PERSON_LASTNAME"
        type="hidden"
        value="Angel"
      />
      <input
        name="PARAMS_OUT"
        type="hidden"
        value="ORDER_NUMBER,PAYMENT_ID,AMOUNT,CURRENCY,PAYMENT_METHOD,TIMESTAMP,STATUS"
      />
      <input
        name="AUTHCODE"
        type="hidden"
        value="94318EF18EAAA5B32A3F5484D8D8893ACFD4C0551BCDFBFDC52F1B7239F0CB1B"
      />
      <button
        type="submit"
      >
        Pay
      </button>
    </form>
  </section>
</div>
`;
